### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by ker0x**

### CODE REVIEW

**Comment by norkunas** [comment]
I think instead of setting `SYMFONY_DEPRECATIONS_HELPER: 'max[self]=3&max[indirect]=1'` we could use deprecations baseline file, would be easier

File: tests/allowed.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+[
```
Comment (norkunas) [inline tests/allowed.json:1]: From my experience this file can change randomly when running on different devices.
I suggest this: https://symfony.com/doc/current/components/phpunit_bridge.html#ignoring-deprecations

File: phpunit.xml.dist [hunk 2]
```diff
@@ -1,28 +1,30 @@
...
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    backupGlobals="false"
+    backupStaticAttributes="false"
+    colors="true" convertErrorsToExceptions="true"
+    convertNoticesToExceptions="true"
+    convertWarningsToExceptions="true"
+    processIsolation="false"
+    stopOnFailure="false"
+    bootstrap="tests/bootstrap.php"
+    xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd"
```
Comment (norkunas) [inline phpunit.xml.dist:13]:
```suggestion
    xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.6/phpunit.xsd"
```
?

File: phpunit.xml.dist [hunk 3]
```diff
@@ -1,28 +1,30 @@
...
-         convertNoticesToExceptions="true"
-         convertWarningsToExceptions="true"
-         processIsolation="false"
-         stopOnFailure="false"
-         bootstrap="tests/bootstrap.php"
+<phpunit
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    backupGlobals="false"
+    backupStaticAttributes="false"
+    colors="true" convertErrorsToExceptions="true"
```
Comment (norkunas) [inline phpunit.xml.dist:7]:
```suggestion
    colors="true"
    convertErrorsToExceptions="true"
```

File: tests/allowed.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        "count": 41
+    },
+    {
+        "message": "Since symfony/framework-bundle 6.4: The \"annotation_reader\" service alias is deprecated without replacement. It is being referenced by the \"routing.loader.attribute\" service.",
+        "count": 41
+    },
+    {
+        "message": "Since symfony/framework-bundle 6.4: The \"annotation_reader\" service alias is deprecated without replacement. It is being referenced by the \"doctrine.orm.metadata.annotation_reader\" alias.",
+        "count": 5
+    }
```
Comment (norkunas) [inline tests/allowed.json:40]: Btw these should be fixed, not ignored ;)

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -105,5 +99,5 @@ jobs:
 
             - name: "Run tests"
               env:
-                  SYMFONY_DEPRECATIONS_HELPER: 'max[self]=3&max[indirect]=1'
+                  SYMFONY_DEPRECATIONS_HELPER: 'baselineFile=./tests/baseline-ignore'
```
Comment (norkunas) [inline .github/workflows/ci.yml:102]:
```suggestion
                  SYMFONY_DEPRECATIONS_HELPER: 'ignoreFile=./tests/baseline-ignore'
```

File: tests/allowed.json (hunk 1 above)
Comment (ker0x) [reply tests/allowed.json:1]: The `ignoreFile` option is only available since Symfony `6.1`!

Comment (norkunas) [reply tests/allowed.json:1]: Ah, sad :(

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -95,7 +92,10 @@ jobs:
...
 
             - name: "Checkout code"
-              uses: actions/checkout@v4
+              uses: actions/checkout@4
+
+            - name: "Configure Symfony"
+              run: composer config extra.symfony.require "${{ matrix.sf_version }}"
 
             - name: "Install Composer dependencies"
               uses: "ramsey/composer-install@v2"
```
Comment (norkunas) [inline .github/workflows/ci.yml:101]:
```suggestion
              uses: "ramsey/composer-install@v2"
              env:
                  SYMFONY_REQUIRE: ${{ matrix.sf_version }}
```

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -95,7 +92,10 @@ jobs:
                   coverage: none
 
             - name: "Checkout code"
-              uses: actions/checkout@v4
+              uses: actions/checkout@4
+
+            - name: "Configure Symfony"
+              run: composer config extra.symfony.require "${{ matrix.sf_version }}"
 
             - name: "Install Composer dependencies"
```
Comment (norkunas) [inline .github/workflows/ci.yml:100]:
```suggestion
              uses: actions/checkout@4

            - name: "Install Composer dependencies"
```

**Review by norkunas — CHANGES_REQUESTED** [review]

File: .editorconfig [hunk 8]
```diff
@@ -7,3 +7,9 @@ indent_size = 4
 indent_style = space
 insert_final_newline = true
 trim_trailing_whitespace = true
+
+[*.json]
+indent_size = 2
+
+[.github/**/*.yml]
+indent_size = 2
```
Comment (norkunas) [inline .editorconfig:15]: not sure what's the point of these changes if PR targets at fixing tests :) only hardens the review

File: tests/allowed.json [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+  {
+    "location": "Bazinga\\GeocoderBundle\\Tests\\Functional\\GeocoderListenerTest::testPersistForEmptyProperty",
+    "message": "Since doctrine/doctrine-bundle 2.11: Not setting \"enable_lazy_ghost_objects\" to true is deprecated.",
+    "count": 1
+  },
+  {
+    "location": "Bazinga\\GeocoderBundle\\Tests\\Functional\\GeocoderListenerTest::testDoesNotGeocodeIfAddressNotChanged",
+    "message": "Since doctrine/doctrine-bundle 2.11: Not setting \"enable_lazy_ghost_objects\" to true is deprecated.",
+    "count": 1
+  },
```
Comment (norkunas) [inline tests/allowed.json:41]: these could be resolved with:
```php
if (method_exists(Configuration::class, 'setLazyGhostObjectEnabled') && Kernel::VERSION_ID >= 60100) {
    $container->prependExtensionConfig('doctrine', [
        'orm' => [
            'enable_lazy_ghost_objects' => true,
        ],
    ]);
}
```

File: .editorconfig (hunk 8 above)
Comment (ker0x) [reply .editorconfig:15]: Every time I modify the `composer.json` or the `ci.yml` my IDE indents the additions with 4 spaces as defined in the .editorconfig whereas they are indented with 2 spaces, these 2 lines save me from having to correct the indentation manually

**Comment by ker0x** [comment]
@norkunas I'm sorry, but I think I'm going to give up because there are far too many deprecations to manage with version 4.4. Every time I fix one, another pops up right after it, it's hellish!

**Comment by norkunas** [comment]
Sorry to hear that :disappointed: I'll try to complete it then

**Comment by ker0x** [comment]
@norkunas Holy Moly, tests are green 🥳 🎉  ! https://github.com/ker0x/geocoder-php-geocoder-bundle/actions/runs/7708347743

**Review by norkunas — APPROVED** [review]
I'm merging as is, but we should probably resolve some deprecations in follow up PR's because there are way too many of them :)

**Comment by norkunas** [comment]
Thank you @ker0x !

**Pull request merged**